### ISSUE TITLE
How do I load a model for inference?

### ISSUE THREAD

**Issue by camhpj** [ultralytics/ultralytics#8944]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I have training a yolov8 model and want to load it for inference. I am loading the model with `model = YOLO('model.pt', task='detect')`. I was under the impression I needed to call `model.eval()` in order to prevent non-deterministic results stemming from batch norm and dropout layers but when I do it starts training on coco8 with the default.yaml config (part of the output is below). Am I misunderstanding something?

```
Ultralytics YOLOv8.1.27 🚀 Python-3.10.12 torch-2.2.1+cu121 CUDA:0 (NVIDIA GeForce RTX 3070, 7959MiB)
engine/trainer: task=detect, mode=train, model=model.pt, data=coco8.yaml, epochs=100, time=None, patience=100, batch=16, imgsz=640, save=True, save_period=-1, cache=False, device=None, workers=8, project=None, name=train, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=True, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, show_boxes=True, line_width=None, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=None, workspace=4, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, mosaic=1.0, mixup=0.0, copy_paste=0.0, auto_augment=randaugment, erasing=0.4, crop_fraction=1.0, cfg=None, tracker=botsort.yaml, save_dir=/runs

Dataset 'coco8.yaml' images not found ⚠️, missing path ''
Downloading https://ultralytics.com/assets/coco8.zip to 'datasets/coco8.zip'...
100%|██████████| 433k/433k [00:00<00:00, 6.93MB/s]
Unzipping datasets/coco8.zip to datasets/coco8...: 100%|██████████| 25/25 [00:00<00:00, 6460.73file/s]
Dataset download success ✅ (0.7s), saved to datasets
```

**Comment by camhpj** [comment]
Thanks for the quick reply! After digging through the code base more I see that `model.eval()` is called by `BasePredictor.setup_model()`. I am working with this code base for the first time and I have to say it is great. Very readable and intuitive given the complexity!

**Comment by glenn-jocher** [comment]
@camhpj, I'm thrilled to hear you're finding the code base readable and intuitive! 😊 It's fantastic that you've dived into the details and discovered how `model.eval()` is incorporated. We've aimed to make YOLOv8 as user-friendly as possible, despite its complex functionalities. If you have any more questions or need further assistance as you explore YOLOv8, don't hesitate to reach out. Happy coding! 🚀
